pkg/sharing/secret_import_reconciler.go

```diff
@@ -237,3 +276,38 @@ func (r *SecretImportReconciler) updateStatus(
 	}
 	return nil
 }
+
+// enqueueNamespaceToSecret is a custom handler that is optimized for
+// tracking Namespace annotation change events. It tries to result in minimum number of
+// Secret reconcile requests.
+type enqueueNamespaceToSecret struct {
+	ToRequests handler.MapFunc
+	Log        logr.Logger
+}
+
+// Create doesn't do anything
+func (e *enqueueNamespaceToSecret) Create(evt event.CreateEvent, q workqueue.RateLimitingInterface) {}
+
+// Update checks whether the exclusion annotation has been added or removed and then touches the secrets in that namespace
+func (e *enqueueNamespaceToSecret) Update(evt event.UpdateEvent, q workqueue.RateLimitingInterface) {
+	typedNsOld, okOld := evt.ObjectOld.(*corev1.Namespace)
+	typedNsNew, okNew := evt.ObjectNew.(*corev1.Namespace)
+	if okOld && okNew && (nsHasExclusionAnnotation(*typedNsOld) == nsHasExclusionAnnotation(*typedNsNew)) {
+		return // Skip when exclusion annotation did not change
+	}
+
+	e.mapAndEnqueue(q, evt.ObjectNew)
+}
+
+// Delete doesn't do anything
+func (e *enqueueNamespaceToSecret) Delete(evt event.DeleteEvent, q workqueue.RateLimitingInterface) {}
+
+// Generic doesn't do anything
+func (e *enqueueNamespaceToSecret) Generic(evt event.GenericEvent, q workqueue.RateLimitingInterface) {
```

Comment:
nit: `{}` formatting

Reply:
hmm. - for the record part of what happens is that some combination of things running in my editor disagree with your formatting preferences.    there's hooks for customizing gofmt, right?

Reply:
like i don't how i got away with line 303 above this,   right now every time i save it just puts the `}` back on the line below